### PR TITLE
Add Unit Tests for acm_certificates_expiration_check

### PR DESCRIPTION
**Pull request by prajwal-choudhari-comprinno**
### **Context**

Adding unit test coverage for the `acm_certificates_expiration_check` implemented under `library/aws/checks/acm/`.
This ensures correctness across various ACM certificate states (expired, expiring soon, valid, no certificates, and errors).

### **Description**

* Created `test_acm_certificates_expiration_check.py` in `library/aws/tests/acm/`
* Test cases added:

  * No ACM certificates
  * Expired certificate
  * Certificate expiring in less than 7 days
  * Valid certificate (more than 7 days remaining)
  * Error while describing certificate
  * Error while listing certificates

Dependencies:

* Relies on `pytest` and `unittest.mock`
* No external AWS resources required due to mocking

### **Checklist**

* [x] Added new tests for existing check
* [x] Verified all tests cover various execution paths and edge cases
* [x] Code covered by tests
* [x] Follows [[Google Python Style Guide](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)](https://github.com/google/styleguide/blob/gh-pages/pyguide.md#38-comments-and-docstrings)
* [x] No backporting required
* [x] Confirmed that mocking avoids actual AWS API calls

### **License**

I confirm that my contribution is made under the terms of the **Apache 2.0 license**.

### CODE REVIEW

File: library/aws/tests/acm/test_acm_certificates_expiration_check.py [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+    CheckMetadata,
+    Remediation,
+    RemediationCode,
+    RemediationRecommendation,
+)
+from library.aws.checks.acm.acm_certificates_expiration_check import acm_certificates_expiration_check
+
+
+@pytest.fixture
+def test_metadata():
```
Comment (aafaq-rashid-comprinno) [inline library/aws/tests/acm/test_acm_certificates_expiration_check.py]: Fix naming conventions

Comment (prajwal-choudhari-comprinno) [reply library/aws/tests/acm/test_acm_certificates_expiration_check.py]: corrected
